javascript/ql/src/semmle/javascript/RangeAnalysis.qll

```diff
@@ -161,31 +161,35 @@ module RangeAnalysis {
     not exists(r.asExpr().getIntValue()) and
     (
       exists (AddExpr expr | r.asExpr() = expr |
+        // r = root + k
```

Comment:
Do you want to deduplicate `sign = 1`?

```
      exists (AddExpr expr | r.asExpr() = expr and sign = 1 |
```
Ditto for `SubExpr`

Reply:
Yes, that was done on purpose.

Reply:
To elaborate a bit, sharing `sign = 1` here would be inconsistent with all the other clauses.

Reply:
I see.